### PR TITLE
Nvtx profiling support

### PR DESCRIPTION
**Pull request by markusbattarbee**
Tested on puhti with:

Currently Loaded Modules:
  1) pgi/19.7   2) hpcx-mpi/2.5.0-cuda   3) cuda/10.1.168

cd src
activate NVTX support in Makefile
make
cd ../example/nvtx_test
make
srun -n1 -p gputest --gres=gpu:v100:1 --account=project_2000203 --time=00:02:30 nvprof -f -o out.nvvp ./test
nvprof -i out.nvvp

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -19,18 +19,28 @@ OUT_SHARED_NO = ../lib/libnophiprof.so
 CLOCK_ID = CLOCK_MONOTONIC
 
 # C++ compiler flags 
-CCFLAGS = -fpic -O2 -fopenmp -std=c++17 -DCLOCK_ID=$(CLOCK_ID) -W -Wall -Wextra -pedantic 
+CCFLAGS = -fpic -O2 -std=c++17 -DCLOCK_ID=$(CLOCK_ID) #-W -Wall -Wextra -pedantic
```
Comment (galfthan) [inline src/Makefile]: Should re-enable warnings. Now that it compiles cleanly we want to make sure it continues to. 

GCC with openmp and no nvtx should be standard. We need a better way (configure ?) for handling different platforms.

File: example/nvtx_test/test.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+   }
+   phiprof::stop("100x0.01s computations");
+
+   if(rank==0)
+      cout << "  2/3" <<endl;
+   MPI_Barrier(MPI_COMM_WORLD);
+
+   phiprof::start("100 x 0.01 (threadId + 1)s, ID");
+   int id = phiprof::initializeTimer("compute");
+#pragma omp parallel
```
Comment (galfthan) [inline example/nvtx_test/test.cpp]: should this test not be with openacc? openmp is not even enabled in makefile

Comment (markusbattarbee) [reply example/nvtx_test/test.cpp]: I thought it would have one openacc parallel loop and also some openmp stuff for completeness.

Comment (galfthan) [reply example/nvtx_test/test.cpp]: Check!

File: src/Makefile (hunk 1 above)
Comment (markusbattarbee) [reply src/Makefile]: Ok, I re-structured the makefile to allow different flags for different compiler environments. NVTX is default on for PGI. Also updated the Readme.

**Comment by markusbattarbee** [comment]
Added checks to work with our clang/cuda environment on Mahti. Profiling there works.

image

@galfthan, any other requests or should this be merged?

**Comment by markusbattarbee** [comment]
Ok @galfthan nvcc support is included and tested now.
Screenshot 2022-06-06 at 11 00 22

**Comment by markusbattarbee** [comment]
We now merged also roctx and roctracer support into this branch.

**Pull request merged**